### PR TITLE
Prevent negative price setting in input

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it prevents placing negative price in inputs.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1243 [Variants] User shouldn't be able to set value below zero using button](https://app.clickup.com/t/2549495/SALEOR-1243)

**Review by AlicjaSzu — APPROVED** [review]

File: src/products/components/ProductPricing/ProductPricing.tsx [hunk 1]
```diff
@@ -37,6 +37,12 @@ const ProductPricing: React.FC<ProductPricingProps> = props => {
 
   const formErrors = getFormErrors(["basePrice"], errors);
 
+  const handlePriceChange = event => {
+    if (/^\d*(\.\d+)?$/.test(event.target.value)) {
```
Comment (dominik-zeglen) [inline src/products/components/ProductPricing/ProductPricing.tsx:41]: As I mentioned it in #754, this should be util func

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Review by dominik-zeglen — APPROVED** [review]

**Comment by orzechdev** [comment]
passed to QA

**Review by mlamperski — APPROVED** [review]
Tested positive

**Pull request merged**